### PR TITLE
Stash improvements - async queues to allow HTTP requests from logger …

### PR DESCRIPTION
**Pull request by whiteh**
…context (#3)

* adding stash improvements - async queue for HTTP requests out of logger context

* setting higher numbers for allowed timers

* error fix

### CODE REVIEW

File: Dockerfile.nginx_lua [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    make . && \
+    make install
+
+# Set up Netacea module
+COPY ./lua_resty_netacea-0.0-2.rockspec ./
+COPY ./src ./src
+RUN luarocks make ./lua_resty_netacea-0.0-2.rockspec
+
+# Link CA certs so they match expected filename
+RUN ln -s /etc/ssl/certs/ca-bundle.crt /etc/ssl/certs/ca-certificates.crt
```
Comment (Scotsoo) [inline Dockerfile.nginx_lua:77]: cheeky

File: src/lua_resty_netacea.lua [hunk 2]
```diff
@@ -352,19 +508,39 @@ function _N:ingest()
...
+  -- start STASH code
+  local request_params = {};
+
+  request_params.body  = cjson.encode(data);
+  request_params.method  = "POST";
+  request_params.headers = {
+        ["Content-Length"] = #request_params.body,
+        ["Content-Type"] = "application/json",
+        ["x-netacea-api-key"] = self.apiKey;
+  }
```
Comment (safderiqbal) [inline src/lua_resty_netacea.lua]:
```suggestion
  };
```

**Review by Scotsoo — APPROVED** [review]

**Review by safderiqbal — APPROVED** [review]
A missing semi-colon but we can forgive that! 😆

File: src/lua_resty_netacea.lua [hunk 3]
```diff
@@ -352,19 +508,39 @@ function _N:ingest()
...
+  request_params.timeout = 1000; -- 1 second
+
+  local request_task = function()
+        local hc = http:new();
```
Comment (Mindez) [inline src/lua_resty_netacea.lua]: There's some very funky indentation throughout, could you run the file through something to fix this?

Comment (Mindez) [reply src/lua_resty_netacea.lua]: Original file was all 2 space, this seems to have some 4 and some 6 in places?

**Review by Mindez — CHANGES_REQUESTED** [review]

**Review by Mindez — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -13,11 +13,12 @@ events {
 }
 
 http {
+  lua_package_path "/usr/local/share/lua/5.1/?.lua;;";
   lua_need_request_body on;
```
Comment (safderiqbal) [inline README.md:20]:
```suggestion
  lua_max_running_timers  2048;
  lua_max_pending_timers  4096;
  lua_socket_pool_size    1024;
  lua_need_request_body on;
```

File: README.md [hunk 5]
```diff
@@ -55,11 +56,12 @@ events {
...
```
Comment (safderiqbal) [inline README.md:66]:
```suggestion
  lua_max_running_timers  2048;
  lua_max_pending_timers  4096;
  lua_socket_pool_size    1024;
  lua_need_request_body on;
```

**Pull request merged**